spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Module.scala

```diff
@@ -38,6 +39,11 @@ object Module {
     CaffeLoader.load[T](model, defPath, modelPath, matchAll)
   }
 
+  def loadCaffeDynamic[T: ClassTag](defPath: String, modelPath: String, matchAll: Boolean = true)(
```

Comment:
rename loadCaffeDynamic to loadCaffe, and mark the old load caffe method as deprecated.